### PR TITLE
Bookstore connect-Api

### PR DESCRIPTION
**Pull request by 1GORDON**
# In this project, I achieved the following

- Connected the React-Redux application to an existing API.
- Created and Removed books from a remote server.
- Used middleware to make async requests to the API.

- Live Link https://bookstore-connect-api.netlify.app/

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      removeBook: (state, action) => current(state).filter((b) => b.item_id !== action.payload),
+      fetchBooks: (state, action) => action.payload,
+    },
+  },
+);
+
+const fetchBooks = async (dispatch) => {
+  const response = await fetch(appUrl);
+  const books = await response.json();
+  const bookList = Object.entries(books).map(([id, [book]]) => (
```
Comment (SajjadAhmad14) [inline src/redux/books/books.js:22]: Book items get removed from the page when the user refreshes the page. Please make sure Server requests are made with react hooks and data comes from the API and gets stored in redux.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Godanah, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/books/books.js (hunk 1 above)
Comment (1GORDON) [reply src/redux/books/books.js:22]: Fixed required changes

**Review by omar-labana — APPROVED** [review]
Hi @1GORDON, 

- _[OPTIONAL]_ Adding a book after removing all books throughs aruntime exception, kindly fix that.
image
 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**